src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -277,53 +278,70 @@ StatusOr<std::vector<std::filesystem::path>> FindHostPathForPIDLibs(
                                 HostPathForPIDPathSearchType::kSearchTypeEndsWith);
 }
 
+using ssl_lib_matcher =
+    std::vector<std::tuple<std::vector<std::string_view>, HostPathForPIDPathSearchType>>;
+
+ssl_lib_matcher kLibSSLMatchers = {
+    {
+        {"libssl.so.1.1", "libcrypto.so.1.1"},
+        HostPathForPIDPathSearchType::kSearchTypeEndsWith,
+    },
+    {
+        {kLibNettyTcnativePrefix},
+        HostPathForPIDPathSearchType::kSearchTypeContains,
+    },
+};
```

Comment:
The goal of this was to allow the bulk of the logic within `AttachOpenSSLUProbesOnDynamicLib` to apply to the optimistic boringssl case -- where `Openssl_version_num` is accessible via `dlopen` or the `RawFptrManager` interface. Since envoy's case is the only other boringssl integration we've investigated, I'm not sure if this the correct abstraction so let me know your thoughts on this.